File: app/src/main/java/com/simon/harmonichackernews/data/GitLabInfo.java
package com.simon.harmonichackernews.data;

public class GitLabInfo {

    public String name;
    public String namespace;
    public String description;
    public String website;
    public String language;
    public String visibility;
    public int stars;
    public int forks;

    public String formatStars() {
        return LinkPreviewFormatUtils.formatCount(stars, "star", "stars");
    }

    public String formatForks() {
        return LinkPreviewFormatUtils.formatCount(forks, "fork", "forks");
    }

    public String formatVisibility() {
        if (visibility == null) {
            return null;
        }

        return visibility.substring(0, 1).toUpperCase() + visibility.substring(1);
    }

    public String getShortenedUrl() {
        return LinkPreviewFormatUtils.shortenUrl(website);
    }
}

File: app/src/main/java/com/simon/harmonichackernews/data/GitLabInfo.kt
package com.simon.harmonichackernews.data

import java.util.Locale

class GitLabInfo {
    var name: String? = null
    var namespace: String? = null
    var description: String? = null
    var website: String? = null
    var language: String? = null
    var visibility: String? = null
    var stars: Int = 0
    var forks: Int = 0

    fun formatStars(): String = LinkPreviewFormatUtils.formatCount(stars, "star", "stars")

    fun formatForks(): String = LinkPreviewFormatUtils.formatCount(forks, "fork", "forks")

    fun formatVisibility(): String? {
        val currentVisibility = visibility ?: return null

        return currentVisibility.substring(0, 1).uppercase(Locale.getDefault()) +
            currentVisibility.substring(1)
    }

    val shortenedUrl: String?
        get() = LinkPreviewFormatUtils.shortenUrl(website)
}
